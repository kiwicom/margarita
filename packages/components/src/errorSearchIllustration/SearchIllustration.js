// @flow

import React from 'react';

const baseImagePath = './static/img/error-search-travel';

export default function ErrorSearchIllustration() {
  return (
    <img
      src={`${baseImagePath}.png`}
      srcSet={`${baseImagePath}.png 1x, ${baseImagePath}@2x.png 2x, ${baseImagePath}@3x.png 3x`}
      style={{ width: '80%', alignSelf: 'center', marginBottom: '20px' }}
    />
  );
}
