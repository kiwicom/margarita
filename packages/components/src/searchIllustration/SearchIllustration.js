// @flow

import React from 'react';
import { defaultTokens } from '@kiwicom/orbit-design-tokens';

/**
 * This component could be extracted to include more illustrations.
 * However, that would require fetching and bundling all the assets for mobile.
 */

const baseImagePath = (resolution: number) =>
  `//images.kiwi.com/illustrations/0x${parseInt(
    defaultTokens.heightIllustrationMedium,
    10,
  ) * resolution}/Boarding.png`;

const RESOLUTIONS = [1, 2, 3];

export default function SearchIllustration() {
  return (
    <img
      src={baseImagePath(1)}
      srcSet={RESOLUTIONS.map(
        resolution => `${baseImagePath(resolution)} ${resolution}x`,
      ).join(',')}
      style={{ width: '80%', alignSelf: 'center', marginBottom: '20px' }}
    />
  );
}
