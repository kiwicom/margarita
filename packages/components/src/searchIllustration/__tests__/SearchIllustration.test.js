// @flow

import * as React from 'react';
import { shallow } from 'react-native-testing-library';

import SearchIllustration from '../SearchIllustration';

it('renders', () => {
  expect(shallow(<SearchIllustration />)).toMatchSnapshot();
});
