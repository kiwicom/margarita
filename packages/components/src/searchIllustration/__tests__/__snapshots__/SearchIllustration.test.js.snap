// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "output": <Component
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "marginBottom": 20,
      }
    }
  />,
}
`;
